src/Data/List/Relation/Binary/Permutation/Propositional/Properties.agda

```diff
@@ -28,9 +30,31 @@ open import Relation.Unary using (Pred)
 open import Relation.Binary
 open import Relation.Binary.PropositionalEquality as ≡
   using (_≡_ ; refl ; cong; cong₂; _≢_; inspect)
+open import Relation.Nullary
 
 open PermutationReasoning
 
+
+------------------------------------------------------------------------
+-- Permutations of empty and singleton lists
+
+module _ {a} {A : Set a} where
+
+  ↭-[]-inv : ∀ {xs : List A} → xs ↭ [] → xs ≡ []
```

Comment:
If we're referring to the other proof as ` ↭-singleton-inv` maybe this one should be called ` ↭-empty-inv` for consistency?